src/http/adapter.js

```diff
@@ -132,21 +132,44 @@ function getActionMetadata( state, resource, actionName, action, meta, resources
 	};
 }
 
+function getHandler( handle ) {
+	if( _.isFunction( handle ) ) {
+		return handle;
+	} else {
+		var list = _.map( handle, function( option ) {
+			if( _.isFunction( option.condition ) ) {
+				return option;
+			} else {
+				return {
+					condition: _.matches( option.condition ),
+					call: option.call
+				};
+			}
+		} );
+		return function( envelope ) {
+			var option = _.find( list, function( option ) {
+				return option.condition( envelope );
+			} );
+			return option.call( envelope );
```

Comment:
@digitalBush hmmm, yeah, so is it a 40x because user submitted something I don't even or a 500 because developer derped it up?